roles/openshift_metrics/tasks/setup_certificate.yaml

```diff
@@ -0,0 +1,58 @@
+---
+- name: generate {{ component }} keys
+  command: >
+    {{ openshift.common.admin_binary }} ca create-server-cert
+    --config={{ mktemp.stdout }}/admin.kubeconfig
+    --key='{{ openshift_metrics_certs_dir }}/{{ component }}.key'
+    --cert='{{ openshift_metrics_certs_dir }}/{{ component }}.crt'
+    --hostnames='{{ hostnames }}'
+    --signer-cert='{{ openshift_metrics_certs_dir }}/ca.crt'
+    --signer-key='{{ openshift_metrics_certs_dir }}/ca.key'
+    --signer-serial='{{ openshift_metrics_certs_dir }}/ca.serial.txt'
+  when: not '{{ openshift_metrics_certs_dir }}/{{ component }}.key'|exists
+
+- name: generate {{ component }} certificate
+  shell: >
+    cat
+    '{{ openshift_metrics_certs_dir | quote }}/{{ component|quote }}.key'
+    '{{ openshift_metrics_certs_dir | quote }}/{{ component|quote }}.crt'
+    > '{{ openshift_metrics_certs_dir | quote }}/{{ component|quote }}.pem'
```

Comment:
I'm not sure if it's any better, but instead of using shell/cat, you could use slurp and copy to create the pem from the contents of the key and cert.